test/unit/util/HeaderUtil.test.ts

```diff
@@ -12,39 +12,35 @@ import {
 describe('HeaderUtil', (): void => {
   describe('#parseAccept', (): void => {
     it('parses empty Accept headers.', async(): Promise<void> => {
-      expect(parseAccept('')).toEqual([]);
+      expect(parseAccept('')).toEqual({});
     });
 
     it('parses Accept headers with a single entry.', async(): Promise<void> => {
-      expect(parseAccept('audio/basic')).toEqual([
-        { range: 'audio/basic', weight: 1, parameters: { mediaType: {}, extension: {}}},
-      ]);
+      expect(parseAccept('audio/basic')).toEqual({
+        'audio/basic': 1,
```

Comment:
Now that I see it, I'm a bit concerned about us dropping this info.
Maybe wee want `ValuePreferences` to be a generic `Map<x, Number>`, like `ValuePreferences<Accept>` or so. @joachimvh?

Reply:
We don't really have a specific issue for this afaik but this is related to #458 (but then for preferences instead of content-type).

Seeing this now this might have been the reason that HeaderUtil didn't return a `ValuePreferences` object, since `HeaderUtil` was just a utility library for completely parsing the accept headers, and `AcceptPreferenceParser` then converted this into a format we actually use, which currently means throwing all the extra parameters info away.

Changing ValuePreferences to be generic could be a solution, although I think type is the only one with possible extra parameters? So we could also have a specific interface for that one. Also we can't use the `Accept` interface as it currently is since that one (specifically the underlying `AcceptHeader` interface) already includes a weight field. Could also have it be `Set<T>`/`T[]` in that case since there already is a `ValuePreference` interface which corresponds to a single entry.

The main issue is that this would be a drastic change since there are many classes that depend on the current structure of the preferences. Increasing the complexity of the object would also make it more annoying for many of those classes as it is often used inline, e.g.: https://github.com/solid/community-server/blob/dc7592ebc4863ce04e3ae140322d6bf1b7c6b416/src/authorization/WebAclAuthorizer.ts#L214

So what I'm saying is that doing this change would drastically increase the complexity of the required changes in this PR.

Reply:
Oh dear.

If we for a second disregard the amount of work for the transition, does the interface make sense or not?

Reply:
Something like what you suggested would probably be fine, but invent a new interface instead of the `Accept` interface since that one is specific for the parsing output. Or change what that interface looks like I guess. Could have a new type which is `string | { range: string, parameters?: { ... } }` and use that as key in the object, value still being the weight. That way calls like the one above would not have to change. It would slightly increase parsing complexity for classes that take preferences as input.

The other solution would be to have a Set/array as mentioned above and put the weight field together with the rest in the preference object. I can't immediately see any specific advantages of this version (or disadvantages), it's mostly a stylistic choice.

Reply:
The orginal reason for having it as an object disappears indeed: the quick key/value check whether a certain content type is supported.

I think we'd need to come up with our own data structure for this one, really.
But not for v1.0 then.

I'm afraid we'd have to close the PR and updated the associated issue?

Reply:
I'll leave it up to you. This PR still does what was requested in the issue, but we'll have to change it again if we ever want to fully support all preference parameters.